test/spmd/test_xla_sharding.py

```diff
@@ -131,15 +137,61 @@ def test_mark_sharding_4d(self):
     z_dim = 2 if self.n_devices >= 4 else 1
     xs.mark_sharding(xt, self._get_mesh((1, 1, z_dim, self.n_devices // z_dim)),
                      (0, 1, 2, 3))
-    annotation = '{devices=[1,1,%d,%d]%s}' % (
-        z_dim, self.n_devices // z_dim,
-        ','.join([str(i) for i in range(self.n_devices)
-                 ])) if self.n_devices > 1 else '{maximal device=0}'
-    self.assertEqual(annotation, torch_xla._XLAC._get_xla_sharding_spec(xt))
+
+    if self.n_devices > 1:
+      annotation = '{devices=[1,1,%d,%d]%s}' % (
+          z_dim, self.n_devices // z_dim, ','.join(
+              [str(i) for i in range(self.n_devices)]))
+      self.assertEqual(annotation, torch_xla._XLAC._get_xla_sharding_spec(xt))
 
     actual = (xt + xt).cpu()
     self.assertTrue(torch.allclose(expected, actual))
 
+  def test_mark_sharding_partial(self):
```

Comment:
Can we also test local_shards under partial replication?

Reply:
+1 yes, we could with the new shards access API now.